pkg/container/qemu_runner.go

```diff
@@ -1221,28 +1221,97 @@ func getAvailableMemoryKB() int {
 
 		for s.Scan() {
 			var n int
-			if nItems, _ := fmt.Sscanf(s.Text(), "MemTotal: %d kB", &n); nItems == 1 {
+			// Try to get MemAvailable first (available on newer kernels)
+			if nItems, _ := fmt.Sscanf(s.Text(), "MemAvailable: %d kB", &n); nItems == 1 {
 				return n
 			}
 		}
+
+		// If MemAvailable is not found, fall back to MemFree + Buffers + Cached
+		// Reset the file position
+		if _, err := f.Seek(0, 0); err != nil {
+			return 0
+		}
+		s = bufio.NewScanner(f)
+		
+		var memFree, buffers, cached int
+		for s.Scan() {
+			if nItems, _ := fmt.Sscanf(s.Text(), "MemFree: %d kB", &memFree); nItems == 1 {
+				continue
+			}
+			if nItems, _ := fmt.Sscanf(s.Text(), "Buffers: %d kB", &buffers); nItems == 1 {
+				continue
+			}
+			if nItems, _ := fmt.Sscanf(s.Text(), "Cached: %d kB", &cached); nItems == 1 {
+				continue
+			}
+		}
+		
+		if memFree > 0 {
+			return memFree + buffers + cached
+		}
 	case "darwin":
-		cmd := exec.Command("sysctl", "hw.memsize")
+		// Use vm_stat to get available memory on macOS
+		cmd := exec.Command("vm_stat")
 		output, err := cmd.CombinedOutput()
 		if err != nil {
 			return mem
 		}
-		outputStr := strings.TrimSpace(string(output))
-		parts := strings.Split(outputStr, ": ")
-		if len(parts) != 2 {
-			return mem
-		}
 
-		memsize, err := strconv.ParseInt(parts[1], 10, 64)
-		if err != nil {
-			return mem
+		// Parse vm_stat output
+		outputStr := string(output)
+		lines := strings.Split(outputStr, "\n")
+		
+		var pageSize int64
+		var pagesFree, pagesInactive, pagesSpeculative int64
+		
+		for _, line := range lines {
+			line = strings.TrimSpace(line)
+			
+			// Get page size
+			if strings.Contains(line, "page size of") {
+				if _, err := fmt.Sscanf(line, "Mach Virtual Memory Statistics: (page size of %d bytes)", &pageSize); err != nil {
+					continue
+				}
+				continue
+			}
+			
+			// Get free pages
+			if strings.HasPrefix(line, "Pages free:") {
+				if _, err := fmt.Sscanf(line, "Pages free: %d.", &pagesFree); err != nil {
+					continue
+				}
+				continue
+			}
+			
+			// Get inactive pages
+			if strings.HasPrefix(line, "Pages inactive:") {
+				if _, err := fmt.Sscanf(line, "Pages inactive: %d.", &pagesInactive); err != nil {
+					continue
+				}
+				continue
+			}
+			
+			// Get speculative pages
```

Comment:
We should avoid this, free+inactive 
Speculative is memory marked as "possible use" so under higher memory pressure situations it will encounter OOM problems